src/AbstractTypedContainerImplementation.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\ServiceManager;
+
+use Psr\Container\ContainerExceptionInterface;
+use Psr\Container\ContainerInterface;
+use Psr\Container\NotFoundExceptionInterface;
+
+/**
+ * @psalm-require-extends ServiceManager
+ */
+abstract class AbstractTypedContainerImplementation implements ContainerInterface
+{
+    /**
+     * Returns true if the container can return an entry for the given identifier.
+     * Returns false otherwise.
+     *
+     * `has($id)` returning true does not mean that `get($id)` will not throw an exception.
+     * It does however mean that `get($id)` will not throw a `NotFoundExceptionInterface`.
+     *
+     * @psalm-param string|class-string $id
+     */
+    public function has(string $id): bool
+    {
+        return $this->hasService($id);
+    }
+
+    /**
+     * Finds an entry of the container by its identifier and returns it.
+     *
+     * @param string $id Identifier of the entry to look for.
+     * @psalm-param string|class-string $id
+     * @throws NotFoundExceptionInterface  No entry was found for **this** identifier.
+     * @throws ContainerExceptionInterface Error while retrieving the entry.
+     * @return mixed Entry.
```

Comment:
I'd say we shouldn't, as the mapping is not guaranteed. As an example, you might map a class name to an alternate instance that implements the same interface, but does not extend the original class name. As such, it would be valid for the plugin manager, but the assertion that it's of the same class or inheritance tree is now invalid.